include/linux/mod_compat.h

```diff
@@ -36,4 +36,20 @@ typedef const struct kernel_param zfs_kernel_param_t;
 typedef struct kernel_param zfs_kernel_param_t;
 #endif
 
+#define	ZMOD_RW 0644
+#define	ZMOD_RD 0444
+
+/* BEGIN CSTYLED */
+#define	INT int
+#define	UINT uint
+#define	ULONG ulong
+#define	LONG long
+#define	STRING charp
+/* END CSTYLED */
+
+#define	ZFS_MODULE_PARAM(scope_prefix, name_prefix, name, type, perm, desc) \
```

Comment:
Can you add a comment above this definition explaining how callers are expected to use this macro.  What are expected/valid values for `scope_prefix`, `name_prefix`, `name`, `type`, `perf`, and `desc`.

It seems like there's an opportunity here to unify the names of these options across platforms.  We have some flexibility on the Linux side as long as we also leave a compatibility alias in place using `module_param_named()`.  Of course, we wouldn't want to have to do this for every option.

Does it make sense to define a couple macro's for the common `name_prefix`s;  i.e. `zfs_`, `zfs_arc_`, `zfs_dbuf_`, etc.  For Linux we simply exposed the global variables as named without imposing any additional structure, how were these organized on FreeBSD?